solr/core/src/test/org/apache/solr/util/TestCircuitBreakers.java

```diff
@@ -66,26 +76,39 @@ protected static void indexDocs() {
   public void tearDown() throws Exception {
     super.tearDown();
     dummyMemBreaker.close();
-    dummyCBManager.close();
   }
 
   @After
   public void after() {
     removeAllExistingCircuitBreakers();
   }
 
-  public void testCBAlwaysTrips() {
+  public void testCBAlwaysTripsWithCorrectCode() {
     removeAllExistingCircuitBreakers();
 
     CircuitBreaker circuitBreaker = new MockCircuitBreaker(true);
 
     h.getCore().getCircuitBreakerRegistry().register(circuitBreaker);
 
-    expectThrows(
-        SolrException.class,
-        () -> {
-          h.query(req("name:\"john smith\""));
-        });
+    List.of(
+            SolrException.ErrorCode.TOO_MANY_REQUESTS.code,
+            SolrException.ErrorCode.SERVICE_UNAVAILABLE.code,
+            SolrException.ErrorCode.BAD_REQUEST.code)
+        .forEach(
+            code -> {
+              if (code != SolrException.ErrorCode.TOO_MANY_REQUESTS.code) {
```

Comment:
setting to "too many requests" should also work, right? could we maybe use `null` to indicate do-not-set?
```suggestion
              if (code != null) {
```